rosbag2_compression/src/rosbag2_compression/zstd_compressor.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+#include "logging.hpp"
+#include "rosbag2_compression/zstd_compressor.hpp"
+
+namespace
+{
+// Increasing the compression level will:
+//   - Increase the time taken to compress
+//   - Decrease the size of the compressed data
+// Setting to zero uses Zstd's default value of 3.
+constexpr const int DEFAULT_ZSTD_COMPRESSION_LEVEL = 1;
+
+std::unique_ptr<char[]> get_input_buffer(
```

Comment:
We tried `rcutils` and decided against it since there's a lot of overhead to setup and the `uint8_t` buffer didn't work well when reading in the uncompressed file. 
Using `unique_ptr<char[]>` keeps things simple for reading in a file and compressing it with Zstd.